### PR TITLE
update the aggregate metric to the latest api

### PR DESCRIPTION
**Pull request by oba11**
## Pull Request Checklist

**Is this in reference to an existing issue?**
No, just noticed that the existing metrics doesnt work any longer

#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [x] Update README with any necessary configuration snippets

- [x] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

#### Purpose

Since the aggregation API change since 0.24, this metrics hasn't been working. This PR updates the metrics plugin to resolve it.

#### Known Compatibility Issues
Wont work with sensu api less than 0.24

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachangelog.com/)
 
 ## [Unreleased]
+- metrics-aggregates.rb: Refactored to support new named aggregates introduced in Sensu 0.24
```
Comment (majormoses) [inline CHANGELOG.md]: Does this break it for clients < 0.24?

**Review by majormoses — COMMENTED** [review]
This looks reasonable to me without looking at the APIs. Can you confirm if this breaks older clients and provide a test (can be a simple gist)?

File: CHANGELOG.md (hunk 1 above)
Comment (oba11) [reply CHANGELOG.md]: Yes, it won't work for [sensu-api 0.23](https://sensuapp.org/docs/0.23/api/aggregates-api.html) and below

**Comment by oba11** [comment]
@majormoses Thanks for reviewing the PR, here is a [gist with the comparison](https://gist.github.com/oba11/6074db50c62b428fa3e6f6c4452b04f1).
Would try to write a test for metric check as well

File: bin/metrics-aggregate.rb [hunk 2]
```diff
@@ -95,27 +94,27 @@ def acquire_checks
...
   def run
+    timestamp = Time.now.to_i
     acquire_checks.each do |check|
-      timestamp, aggregate = get_aggregate(check[:check])
-      puts "#{check[:check]} aggregates: #{aggregate}" if config[:debug]
+      aggregate = get_aggregate(check['name'])
+      puts "#{check['name']} aggregates: #{aggregate}" if config[:debug]
       aggregate.each do |result, count|
-        output "#{config[:scheme]}.#{check[:check]}.#{result}", count, timestamp
+        if count.is_a?(Hash)
```
Comment (oba11) [inline bin/metrics-aggregate.rb:111]: @majormoses the only hack is here so that the new endpoint doesnt screw up the existing datapoints

Comment (majormoses) [reply bin/metrics-aggregate.rb:111]: can you add a comment on the line above as why it's needed so that we can remove it at some point in the future.

Comment (majormoses) [reply bin/metrics-aggregate.rb:111]: basically something like:
```
# in 0.23 they changed the api results for aggregates this helps maintain backwards compatibility
if count.is_a?(Hash)
```

Comment (majormoses) [reply bin/metrics-aggregate.rb:111]: Have not heard back from you on this, I am gonna add this comment on release.

**Review by majormoses — APPROVED** [review]
LGTM

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-sensu/versions/2.1.1

**Comment by oba11** [comment]
@majormoses sorry I wasnt checking my github emails for the last few days.
Thanks alot for approving the PR as well.

**Pull request merged**